packages/project/src/merge/merge.ts

```diff
@@ -18,28 +20,48 @@ type Options = {
  * target, but the UUIDs of the source
  */
 // TOOD what if a workflow is removed from the target?
-export function merge(source: Project, target: Project, options) {
+export function merge(
+  source: Project,
+  target: Project,
+  options: Options = { workflowMappings: {} }
+) {
   const finalWorkflows: Workflow[] = [];
-  // TODO: a new workflow in the source will not be handled
-  for (const workflow of target.workflows) {
-    const sourceWorkflow = source.getWorkflow(workflow.id);
-    if (sourceWorkflow) {
-      const mappings = mapUuids(sourceWorkflow, workflow);
-      finalWorkflows.push(mergeWorkflows(sourceWorkflow, workflow, mappings));
-    } else finalWorkflows.push(workflow);
+  const usedTargetIds = new Set<string>();
+
+  for (const sourceWorkflow of source.workflows) {
+    const mappedTargetId = options.workflowMappings?.[sourceWorkflow.id];
+    const targetWorkflow = mappedTargetId
+      ? target.getWorkflow(mappedTargetId)
+      : target.getWorkflow(sourceWorkflow.id);
+
+    if (targetWorkflow) {
+      usedTargetIds.add(targetWorkflow.id);
+      const mappings = mapUuids(sourceWorkflow, targetWorkflow);
+      finalWorkflows.push(
+        mergeWorkflows(sourceWorkflow, targetWorkflow, mappings)
+      );
+    } else {
+      finalWorkflows.push(sourceWorkflow);
+    }
   }
 
-  const mergedProject = new Project(
-    { ...source, ...target, workflows: finalWorkflows },
-    // TODO probably just preserve target repo?
-    { ...source.repo, ...target.repo }
+  // workflows from target that didn't get merged
+  for (const targetWorkflow of target.workflows) {
```

Comment:
Yeah need to be careful here because we need allow workflows to be removed

I would say:

- When merging everything, then delete all target workflows and only keep the source workflows
- When merging partially, allow a null value to be an explicit delete - but otherwise ignore the other workflows

Reply:
So the options are 3.

1. keep all unmapped workflows (both main and sandbox)
2. keep unmapped workflows in target (main)
3. keep unmapped workflow in source (sandbox)